controllers/pruner/pruner_controller.go

```diff
@@ -0,0 +1,361 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	dwv2 "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+	controllerv1alpha1 "github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"github.com/devfile/devworkspace-operator/pkg/conditions"
+	"github.com/devfile/devworkspace-operator/pkg/config"
+	"github.com/devfile/devworkspace-operator/pkg/infrastructure"
+
+	"github.com/operator-framework/operator-lib/prune"
+	"github.com/robfig/cron/v3"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// DevWorkspacePrunerReconciler reconciles DevWorkspace objects for pruning purposes.
+type DevWorkspacePrunerReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+
+	cron *cron.Cron
+}
+
+func shouldReconcileOnUpdate(e event.UpdateEvent, log logr.Logger) bool {
+	log.Info("DevWorkspaceOperatorConfig update event received")
+	oldConfig, ok := e.ObjectOld.(*controllerv1alpha1.DevWorkspaceOperatorConfig)
+	if !ok {
+		return false
+	}
+	newConfig, ok := e.ObjectNew.(*controllerv1alpha1.DevWorkspaceOperatorConfig)
+	if !ok {
+		return false
+	}
+
+	oldCleanup := oldConfig.Config.Workspace.CleanupCronJob
+	newCleanup := newConfig.Config.Workspace.CleanupCronJob
+
+	differentBool := func(a, b *bool) bool {
+		switch {
+		case a == nil && b == nil:
+			return false
+		case a == nil || b == nil:
+			return true
+		default:
+			return *a != *b
+		}
+	}
+	differentInt32 := func(a, b *int32) bool {
+		switch {
+		case a == nil && b == nil:
+			return false
+		case a == nil || b == nil:
+			return true
+		default:
+			return *a != *b
+		}
+	}
+
+	if oldCleanup == nil && newCleanup == nil {
+		return false
+	}
+	if (oldCleanup == nil && newCleanup != nil) || (oldCleanup != nil && newCleanup == nil) {
+		return true
+	}
+	if differentBool(oldCleanup.Enable, newCleanup.Enable) {
+		return true
+	}
+	if differentBool(oldCleanup.DryRun, newCleanup.DryRun) {
+		return true
+	}
+	if differentInt32(oldCleanup.RetainTime, newCleanup.RetainTime) {
+		return true
+	}
+	return oldCleanup.Schedule != newCleanup.Schedule
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *DevWorkspacePrunerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	log := r.Log.WithName("setupWithManager")
+	log.Info("Setting up DevWorkspacePrunerReconciler")
+
+	configPredicate := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return shouldReconcileOnUpdate(e, log)
+		},
+		CreateFunc:  func(e event.CreateEvent) bool { return true },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return true },
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+	}
+
+	maxConcurrentReconciles, err := config.GetMaxConcurrentReconciles()
```

Comment:
It looks like `MaxConcurrentReconciles` creates new goroutines that run for the lifetime of the controller:
https://github.com/kubernetes-sigs/controller-runtime/blob/6ad5c1dd4418489606d19dfb87bf38905b440561/pkg/internal/controller/controller.go#L197-L205

Maybe we should remove this so that the [default](https://github.com/kubernetes-sigs/controller-runtime/blob/6ad5c1dd4418489606d19dfb87bf38905b440561/pkg/internal/controller/controller.go#L319) is used, since we are only reconciling one object now?